src/network/networkd-network.c

```diff
@@ -281,6 +281,18 @@ static int network_load_one(Manager *manager, const char *filename) {
         if (network->ip_masquerade)
                 network->ip_forward |= ADDRESS_FAMILY_IPV4;
 
+        /* set IgnoreCarrierGainLoss=false for DHCP or no static address cases */
+        if (network->ignore_carrier_gainloss) {
+                if (network->dhcp != ADDRESS_FAMILY_NO) {
+                        log_warning("Will not ignore carrier gain or loss when DHCP is configured.");
+                        network->ignore_carrier_gainloss = false;
```

Comment:
That will blow up in fun ways if both DHCP and static addresses are configured, and it's a side-effect that varies based on what other options are set. I don't know what the policy is in systemd-networkd for this, but it sounds like a bad idea.

If I set both DHCP and static addresses, you'd never ignore carrier changes, and potentially drop all the config anyway.

IMO, ignoring carrier changes should do just that, it doesn't matter if DHCP is enabled or not, and if there are static addresses or not. You set the option, it will do what is advertised, and you get to keep all the pieces if it doesn't work for what you wanted to do. I disagree about second-guessing what the user sets in configuration.

Reply:
Haha! Ok this is getting funny. I didnt want to add that piece of code either. But I was forced to add it by @ssahani because he wanted me to address only static IP address cases and not touch DHCP. (Please read his comments way back in 2017. May be hidden under code reviews)

I mentioned same yesterday too in one of my comment. Even I am of the opinion that "You set the option, it will do what is advertised"

Once again, I can revise the PR but only if I get some hint that my PR is in right direction and this feature stands a chance of implementation.

PS: I have been mentioning @ssahani a lot these days but nothing personal against him.